app/screens/ProductList/ProductList.js

```diff
@@ -3,141 +3,95 @@ import { Text, View, Image, TouchableHighlight } from 'react-native';
+    },
+    {
+        key: 5,
+        title: 'Product 5',
+        image: iconImages.jam,
+        description: 'Product 5 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.'
+    },
+    {
+        key: 6,
+        title: 'Product 6',
+        image: iconImages.salami,
+        description: 'Product 6 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.'
+    },
+    {
+        key: 7,
+        title: 'Product 7',
+        image: iconImages.snitch,
+        description: 'Product 7 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.'
+    },
+    {
+        key: 8,
+        title: 'Product 8',
+        image: iconImages.tudorRose,
+        description: 'Product 8 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.'
+    }
+];
 
 class ProductList extends Component {
+    redirect(title, image, description) {
+        this.props.navigation.navigate('Product', {
+            title,
+            image,
+            description
+        });
+    }
+
+    renderItems = () => {
+        return data.map(({key, title, image, description}) => (
+            <View style={styles.productItem}
+                    key={key}>
+                <View style={styles.productInfo}>
+                    <Image
+                        source={image}
+                        style={styles.productImage}/>
+                    <Text style={styles.productTitle}>{title}</Text>
+                </View>
+                <TouchableHighlight
+                    onPress={this.redirect.bind(this, title, image, description)}
```

Comment:
you also may use arrow function instead of bind.
```
    redirect = (title, image, description) =>  {
return () => {

        this.props.navigation.navigate('Product', {
            title,
            image,
            description
        });
}
    }
```